spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -135,6 +135,8 @@ public PubSubSubscriberTemplate pubSubSubscriberTemplate(SubscriberFactory subsc
 			ObjectProvider<PubSubMessageConverter> pubSubMessageConverter) {
 		PubSubSubscriberTemplate pubSubSubscriberTemplate = new PubSubSubscriberTemplate(subscriberFactory);
 		pubSubMessageConverter.ifUnique(pubSubSubscriberTemplate::setMessageConverter);
+		pubSubSubscriberTemplate.setAckExecutor(
+				Executors.newFixedThreadPool(this.gcpPubSubProperties.getSubscriber().getAcknowledgementThreads()));
```

Comment:
We prefer to use `ThreadPoolTaskExecutor` from Spring.
Also you have to consider to make it as a bean to let its lifecycle to work properly, including automatic `shutdown()`.

Reply:
I did not realize destroy method inference existed, with Spring automatically calling close()/shutdown() methods.This was cool! Now I don't have to write my own `@PreDestroy` in autoconfig.

I think the only place it's documented is `@Bean` `destroyMethod() `[javadoc](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Bean.html#destroyMethod--).

Reply:
I don't think so: https://docs.spring.io/spring/docs/5.1.3.RELEASE/spring-framework-reference/core.html#beans-factory-lifecycle

But yes: to have it automatic you need to declare `ThreadPoolTaskExecutor` as a `@Bean`.